grpc-circuitbreaker-utils/src/main/java/org/hypertrace/circuitbreaker/grpcutils/CircuitBreakerConfigProvider.java

```diff
@@ -0,0 +1,86 @@
+package org.hypertrace.circuitbreaker.grpcutils;
+
+import com.typesafe.config.Config;
+import java.util.Map;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class CircuitBreakerConfigProvider {
+
+  public static final String DEFAULT_CONFIG_KEY = "default";
+
+  // Whether to enable circuit breaker or not.
+  private static final String ENABLED = "enabled";
+
+  // Percentage of failures to trigger OPEN state
+  private static final String FAILURE_RATE_THRESHOLD = "failureRateThreshold";
+  // Percentage of slow calls to trigger OPEN state
+  private static final String SLOW_CALL_RATE_THRESHOLD = "slowCallRateThreshold";
+  // Define what a "slow" call is
+  private static final String SLOW_CALL_DURATION_THRESHOLD = "slowCallDurationThreshold";
+  // Number of calls to consider in the sliding window
+  private static final String SLIDING_WINDOW_SIZE = "slidingWindowSize";
+  // Time before retrying after OPEN state
+  private static final String WAIT_DURATION_IN_OPEN_STATE = "waitDurationInOpenState";
+  // Minimum calls before evaluating failure rate
+  private static final String MINIMUM_NUMBER_OF_CALLS = "minimumNumberOfCalls";
+  // Calls allowed in HALF_OPEN state before deciding to
+  // CLOSE or OPEN again
+  private static final String PERMITTED_NUMBER_OF_CALLS_IN_HALF_OPEN_STATE =
+      "permittedNumberOfCallsInHalfOpenState";
+  private static final String SLIDING_WINDOW_TYPE = "slidingWindowType";
+
+  // Global flag for circuit breaker enablement
+  private boolean circuitBreakerEnabled = false;
+  private Map<String, CircuitBreakerConfiguration> circuitBreakerConfigurationMap;
+
+  public CircuitBreakerConfigProvider(Config config) {
+    this.initialize(config);
+  }
+
+  /** Checks if Circuit Breaker is globally enabled. */
+  public boolean isCircuitBreakerEnabled() {
+    return circuitBreakerEnabled;
+  }
+
+  private void initialize(Config circuitBreakerConfig) {
+    circuitBreakerEnabled =
+        circuitBreakerConfig.hasPath(ENABLED) && circuitBreakerConfig.getBoolean(ENABLED);
+    this.circuitBreakerConfigurationMap =
+        circuitBreakerConfig.root().keySet().stream()
+            .filter(key -> !key.equals(ENABLED)) // Ignore the global enabled flag
+            .collect(
+                Collectors.toMap(
+                    key -> key, // Circuit breaker key
```

Comment:
It seems like you're imagining all circuit breakers are siblings within one map - What's the benefit of that? IMO we should define some default json format that we can parse into a config, or let callers build their own programatically. Identical to how guava cache config works actually (although we usually use a custom format and programmatically translate it, the point is you can do it either way - this parser is just a convenience for uniformity).